### PR TITLE
[V1] Refactor model executable interface for all text-only language models

### PR DESCRIPTION
**Pull request by ywang96**
This PR refactors the interface of all **text-only decoder** language models for V1 VLM re-arch and `torch.compile` support. In particular, all model implementations on vLLM will need to meet the following requirements:
- `get_input_embeddings(input_ids)` implemented in `XYZModel`
-  `get_input_embeddings(input_ids)` implemented in `XYZForCausalLM`, `XYZForConditionalGeneration`, and/or `XYZForClassification`
-  `inputs_embeds` is a required (with default `None`) parameter in the `forward` function signature of `XYZModel` and `XYZForCausalLM`, `XYZForConditionalGeneration`, and/or `XYZForClassification`

This PR is a prerequisite of applying #9871 to all multimodal models on vLLM.

### CODE REVIEW

**Review by DarkLight1337 — APPROVED** [review]
LGTM!

After you refactor the multi-modal models, can you update the base interface for vLLM models (`class VllmModel`) to define `get_input_embeddings` and also update the required arguments of `forward`, then add a test to ensure that all of our models implement it?

**Comment by ywang96** [comment]
> LGTM!
> 
> After you refactor the multi-modal models, can you update the base interface for vLLM models (`class VllmModel`) to define `get_input_embeddings` and also update the required arguments of `forward`, then add a test to ensure that all of our models implement it?

@DarkLight1337 Thanks for the review! Yes eventually if we're going to enforce this interface, then I will update those model interfaces once the refactoring is done for all related models.

I also want others to take a look at this PR before we merge it so I disabled auto-merge for now.

**Review by WoosukKwon — APPROVED** [review]
LGTM. Thanks for doing this!

File: vllm/model_executor/models/gpt_bigcode.py [hunk 1]
```diff
@@ -218,18 +218,22 @@ def __init__(self, *, vllm_config: VllmConfig, prefix: str = ""):
...
         kv_caches: List[torch.Tensor],
         attn_metadata: AttentionMetadata,
         intermediate_tensors: Optional[IntermediateTensors],
+        inputs_embeds: Optional[torch.Tensor] = None,
     ) -> Union[torch.Tensor, IntermediateTensors]:
         if get_pp_group().is_first_rank:
-            inputs_embeds = self.wte(input_ids)
-            position_embeds = self.wpe(position_ids)
-            hidden_states = inputs_embeds + position_embeds
+            if inputs_embeds is not None:
```
Comment (DarkLight1337) [inline vllm/model_executor/models/gpt_bigcode.py:234]:
```suggestion
            if inputs_embeds is None:
```

**Pull request merged**